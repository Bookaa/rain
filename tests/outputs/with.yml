!program;1
stmts:
- !assn;1
  let: false
  lhs: !name;1
    value: ctx0
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: Initialization
        catch: false
        func: !name;1
          value: print
      - !call;1
        args: []
        catch: false
        func: !name;1
          value: block
      - !call;1
        args:
        - !str;1
          value: Clean up
        catch: false
        func: !name;1
          value: print
    params:
    - block
- !assn;1
  let: false
  lhs: !name;1
    value: ctx1
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: Initialization
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: One
        catch: false
        func: !name;1
          value: block
      - !call;1
        args:
        - !str;1
          value: Two
        catch: false
        func: !name;1
          value: block
      - !call;1
        args:
        - !str;1
          value: Clean up
        catch: false
        func: !name;1
          value: print
    params:
    - block
- !assn;1
  let: false
  lhs: !name;1
    value: ctx2
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: Initialization
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: One
        - !str;1
          value: '1'
        catch: false
        func: !name;1
          value: block
      - !call;1
        args:
        - !str;1
          value: Two
        - !str;1
          value: '2'
        catch: false
        func: !name;1
          value: block
      - !call;1
        args:
        - !str;1
          value: Clean up
        catch: false
        func: !name;1
          value: print
    params:
    - block
- !assn;1
  let: false
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;1
      stmts:
      - !with;1 {}
      - !with;1 {}
      - !with;1 {}
    params: []
