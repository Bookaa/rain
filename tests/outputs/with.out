Initialization
Anonymous block
Clean up
Initialization
Anonymous block: One
Anonymous block: Two
Clean up
Initialization
Anonymous block: One + 1
Anonymous block: Two + 2
Clean up
